scripts/showcase/samplesheet-prep-test.py

```diff
@@ -1,55 +1,399 @@
+    :return:
+    """
+    # Initialise logger
+    # set up logging to file - see previous section for more details
+    logging.basicConfig(level=logging.DEBUG,
+                        format=LOGGER_STYLE,
+                        datefmt=LOGGER_DATEFMT)
+    # define a Handler which writes INFO messages or higher to the sys.stderr
+    console = logging.StreamHandler()
+    console.setLevel(logging.INFO)
+    # set a format which is simpler for console use
+    formatter = logging.Formatter(CONSOLE_LOGGER_STYLE)
+    # tell the handler to use this format
+    console.setFormatter(formatter)
+    # add the handler to the root logger
+    #logging.getLogger('').addHandler(console)
+
+
+def get_logger():
+    """
+    Get the name of where this function was called from - return a logging object
+    Use a trackback from the inspect to do this
+    :return:
+    """
+    return logging.getLogger(THIS_SCRIPT_NAME)
+
+
+# Get logger
+initialise_logger()
+logger = get_logger()
+
+
+def get_args():
+    """
+    Read args
+    return: argparse dict
+    """
+    # Initialise parser
+    parser = argparse.ArgumentParser()
+    # Place arguments
+    parser.add_argument("--samplesheet", "-s", type=str, required=True)
+    parser.add_argument("--trackingsheet", "-t", type=str, required=True)
+    parser.add_argument("--outputFile", "-o", type=str, required=True)
+
+    return parser.parse_args()
+
+
+def check_args(args):
+    """
+    Check each of the arguments are legit
+    param args: argparse input
+        attrs:
+            samplesheet
+            trackingsheet
+            outputFile
+    """
+
+    # Check samplesheet is a real file
+    if not Path(args.samplesheet).is_file():
+        logger.info("Sample sheet does not exist, exiting")
+        sys.exit(1)
+
+    # Check tracksheet is a real file
+    if not Path(args.trackingsheet).is_file():
+        logger.info("Tracking sheet does not exist, exiting")
+        sys.exit(1)
+
+    # Check output file is writable
+    output_dir = os.path.dirname(os.path.normpath(args.outputFile))
+    if not Path(output_dir).is_dir():
+        logger.info("Creating output folder")
+        create_output_dir(output_dir)
+
+
+def create_output_dir(output_dir):
+    """
+    param: output_dir: path to the output directory
+    """
+    # create output directory if it does not exist
+    output_dir = os.path.dirname(output_dir)
+    os.makedirs(output_dir, exist_ok=True)
+
+
+def get_start_row(sample_sheet):
+    """
+    Find on which line [Data] exists.
+    :param sample_sheet: path to the samplesheet
+    :return: An integer specifying which row the [Data] header is on.
+    """
+
+    backslash_chars = '[]'
+    escaped_header_line = ''.join(["\\%s" % char if char in backslash_chars else char
+                                   for char in HEADER_LINE_PRECURSOR
+                                  ])
+
+    re_pattern = re.compile("^%s(,)*" % escaped_header_line)
```

Comment:
We can possibly get rid of this part, as discussed

Reply:
https://github.com/umccr/infrastructure/pull/16/files#diff-9803ecbbdf15c05af6e26ccc75f88c84R72-R74

Reply:
https://github.com/umccr/infrastructure/pull/16/files#diff-9803ecbbdf15c05af6e26ccc75f88c84R122-R129

Reply:
Yep we're using startswith("[Data]") now.